src/api/js/scripts/parse-api.ts

```diff
@@ -32,35 +31,49 @@ let primitiveTypes = {
   int64_t: 'bigint',
   double: 'number',
   float: 'number',
-};
+} as unknown as Record<string, string>;
 
-let optTypes = {
+const optTypes = {
   __proto__: null,
 
   Z3_sort_opt: 'Z3_sort',
   Z3_ast_opt: 'Z3_ast',
   Z3_func_interp_opt: 'Z3_func_interp',
-};
+} as unknown as Record<string, string>;
 
 // parse type declarations
-let types = {
-    __proto__: null,
-
-    // these are function types I can't be bothered to parse
-    Z3_error_handler: 'Z3_error_handler',
-    Z3_push_eh: 'Z3_push_eh',
-    Z3_pop_eh: 'Z3_pop_eh',
-    Z3_fresh_eh: 'Z3_fresh_eh',
-    Z3_fixed_eh: 'Z3_fixed_eh',
-    Z3_eq_eh: 'Z3_eq_eh',
-    Z3_final_eh: 'Z3_final_eh',
-    Z3_created_eh: 'Z3_created_eh',
-    Z3_decide_eh: 'Z3_decide_eh'
-};
+const types = {
+  __proto__: null,
 
-let defApis = Object.create(null);
-let functions = [];
-let enums = Object.create(null);
+  // these are function types I can't be bothered to parse
```

Comment:
this comment predates the macro Z3_DECLARE_CLOSURE.
You can use Z3_DECLARE_CLOSURE to initialize this table.